### PR TITLE
Fix for #994 : a failure during wallet restoration is caught now

### PR DESCRIPTION
**Pull request by kushti**
This PR is fixing issue #994 (If an exception is happening during /wallet/restore (due to e.g. no strong crypto license on Java 8) then the wallet actor is dying).

Logging is also improved for /wallet/init & /wallet/restore

### CODE REVIEW

File: src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala [hunk 1]
```diff
@@ -183,8 +184,12 @@ class ErgoWalletActor(settings: ErgoSettings, boxSelector: BoxSelector)
       storage.updateStateContext(s.stateContext)
   }
 
-  private def walletCommands: Receive = {
-    case InitWallet(pass, mnemonicPassOpt) if secretStorageOpt.isEmpty && walletSettings.testMnemonic.isEmpty =>
+  //Secret is set in form of keystore file of testMnemonic in the config
+  private def secretSet: Boolean = secretStorageOpt.nonEmpty || walletSettings.testMnemonic.nonEmpty
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala]: Consider renaming `secretIsSet` or `secretHasBeenSet`

File: src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala [hunk 2]
```diff
@@ -194,27 +199,44 @@ class ErgoWalletActor(settings: ErgoSettings, boxSelector: BoxSelector)
...
+          self ! UnlockWallet(pass)
+          util.Arrays.fill(entropy, 0: Byte)
+          log.info("Wallet is initialized")
+          s
+        case Failure(t) =>
+          val f = wrapLegalExc(t) //getting nicer message for illegal key size exception
+          log.error(s"Wallet initialization is failed, details: ${f.exception.getMessage}")
+          f
+      }
       sender() ! mnemonicTry
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala]:
```suggestion
      sender() ! mnemonicTry  // TODO refactor: introduce a message class which will reflect the logic 
```

File: src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala [hunk 3]
```diff
@@ -194,27 +199,44 @@ class ErgoWalletActor(settings: ErgoSettings, boxSelector: BoxSelector)
...
+        case s: Success[Unit] =>
+          self ! UnlockWallet(encryptionPass)
+          log.info("Wallet is restored")
+          s
+        case Failure(t) =>
+          val f = wrapLegalExc(t) //getting nicer message for illegal key size exception
+          log.error(s"Wallet restoration is failed, details: ${f.exception.getMessage}")
+          f
+      }
+      sender() ! res
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala]:
```suggestion
      sender() ! res // TODO refactor: introduce a message class which will reflect the logic 
```

File: src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala [hunk 4]
```diff
@@ -194,27 +199,44 @@ class ErgoWalletActor(settings: ErgoSettings, boxSelector: BoxSelector)
...
 
+    // branch for key already being set
     case _: RestoreWallet | _: InitWallet =>
       sender() ! Failure(new Exception("Wallet is already initialized or testMnemonic is set. Clear current secret to re-init it."))
```
Comment (aslesarenko) [inline src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala]:
```suggestion
      // TODO refactor: introduce a message class which will reflect the logic 
      sender() ! Failure(new Exception("Wallet is already initialized or testMnemonic is set. Clear current secret to re-init it."))
```

File: src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala (hunk 2 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala]: https://github.com/ergoplatform/ergo/issues/1005

File: src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala (hunk 3 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala]: https://github.com/ergoplatform/ergo/issues/1005

File: src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala (hunk 4 above)
Comment (kushti) [reply src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala]: https://github.com/ergoplatform/ergo/issues/1005

**Pull request merged**